alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -286,23 +291,27 @@ class TransportIndexMonitorAction @Inject constructor(
                 return
             }
 
-            if (request.method == RestRequest.Method.PUT) return updateMonitor()
-
-            val query = QueryBuilders.boolQuery().filter(QueryBuilders.termQuery("${Monitor.MONITOR_TYPE}.type", Monitor.MONITOR_TYPE))
-            val searchSource = SearchSourceBuilder().query(query).timeout(requestTimeout)
-            val searchRequest = SearchRequest(SCHEDULED_JOBS_INDEX).source(searchSource)
-            client.search(
-                searchRequest,
-                object : ActionListener<SearchResponse> {
-                    override fun onResponse(searchResponse: SearchResponse) {
-                        onSearchResponse(searchResponse)
-                    }
+            if (request.method == RestRequest.Method.PUT) {
+                scope.launch {
```

Comment:
Other monitors are using a different scope to launch coroutine - https://github.com/opensearch-project/alerting/pull/424/files#diff-3e3ee43a4c252b4b9f3513cfa335a2d5ecc79efe161bd0cc0b5cd56b5c14fddeR56 
I'm not sure which one would be more efficient, so give some thought to it if you haven't already.

Reply:
I will refactor this class in a future PR to make it all aligned.